ticketutil/bugzilla.py

```diff
@@ -0,0 +1,367 @@
+
+DEBUG = False
+
+if DEBUG:
+    logging.basicConfig(level=logging.DEBUG)
+else:
+    logging.basicConfig(level=logging.INFO)
+
+
+class BugzillaTicket(ticket.Ticket):
+    """
+    A BZ Ticket object. Contains BZ-specific methods for working with tickets.
+    """
+    def __init__(self, url, project, auth=None, ticket_id=None):
+        self.ticketing_tool = 'Bugzilla'
+
+        # Kerberos is default auth if the auth param is not specified.
+        # A tuple of the form (<username>, <password>) can also be passed in.
+        if isinstance(auth, tuple):
+            self.auth = auth
+            username, password = auth
+            self.credentials = {"login": username, "password": password}
+        else:
+            self.auth = 'kerberos'
+        self.token = None
+
+        # BZ URLs
+        self.url = url
+        self.rest_url = '{0}/rest/bug'.format(self.url)
+        self.auth_url = '{0}/rest/login'.format(self.url)
+
+        # Call our parent class's init method which creates our requests session.
+        super(BugzillaTicket, self).__init__(project, ticket_id)
+
+    def _generate_ticket_url(self):
+        """
+        Generates the ticket URL out of the url, project, and ticket_id.
+        :return: ticket_url: The URL of the ticket.
+        """
+        ticket_url = None
+
+        # If we are receiving a ticket_id, set ticket_url.
+        if self.ticket_id:
+            ticket_url = "{0}/show_bug.cgi?id={1}".format(self.url, self.ticket_id)
+
+        return ticket_url
+
+    def _create_requests_session(self):
+        """
+        Returns to the super class if the authentication method is kerberos.
+        Creates a Requests Session and authenticates to base API URL with authentication other then kerberos.
+        We're using a Session to persist cookies across all requests made from the Session instance.
+        :return s: Requests Session.
+        """
+        if self.auth == 'kerberos':
+            return super(BugzillaTicket, self)._create_requests_session()
+
+        # If basic authentication is passed in, generate a token to be used in all requests.
+        try:
+            s = requests.Session()
+            r = s.get(self.auth_url, params=self.credentials, verify=False)
+            r.raise_for_status()
+            resp = r.json()
+            self.token = resp['token']
+            logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
+            logging.info("Successfully authenticated to {0}.".format(self.ticketing_tool))
+            return s
+
+        # We log an error if authentication was not successful, because rest of the HTTP requests will not succeed.
+        # If authentication wasn't successful, a token will not be in resp. Add KeyError as exception.
+        except (KeyError, requests.RequestException) as e:
+            logging.error("Error authenticating to {0}. No valid credentials were provided.".format(self.auth_url))
```

Comment:
It might be beneficial to have something returned to indicate failure to anything calling this method. As it is, it would only log the error, but anything calling it would continue - which may not be what you want.